lib/internal/modules/esm/shared_constants.js

```diff
@@ -3,6 +3,40 @@
 
 'use strict';
 
+/*
+The shared memory area is divided in 4 32-bit long sections. It has to be 32-bit
+long as `Atomics.notify` only works with `Int32Array` objects.
+
+--32-bits-- --32-bits-- --32-bits-- --32-bits--
+    ^           ^           ^           ^
+    |           |           |           |
+    |           |           | NUMBER_OF_INCOMING_ASYNC_RESPONSES
+    |           |           |
+    |           | NUMBER_OF_INCOMING_MESSAGES
+    |           |
+    |   BOREDOM_OF_THREADS
+    |
+WORKER_TO_MAIN_THREAD_NOTIFICATION
+
+WORKER_TO_MAIN_THREAD_NOTIFICATION is only used to send notifications, its value is never used.
+
+BOREDOM_OF_THREADS does not represent a number, but an array of booleans.
+Currently, only two flags are used: WORKER_THREAD_IS_BORED, and MAIN_THREAD_IS_BORED
```

Comment:
I feel like “idle” is the more commonly word used to describe this, rather than “bored”?